File: app/src/main/java/com/ctyeung/popularmoviestage2/utilities/JSONhelper.java
package com.ctyeung.popularmoviestage2.utilities;

import org.json.JSONArray;
import org.json.JSONObject;

/**
 * Created by ctyeung on 8/18/17.
 */

public class JSONhelper
{
    public static JSONObject parseJson(String str)
    {
        JSONObject json = null;

        try
        {
            json = new JSONObject(str);
        }
        catch (Exception ex)
        {
            ex.printStackTrace();
        }
        return json;
    }

    public static JSONArray getJsonArray(JSONObject json, String key)
    {
        JSONArray jsonArray = null;
        try
        {
            jsonArray = json.getJSONArray("results");
        }
        catch (Exception ex)
        {
            ex.printStackTrace();
        }
        return jsonArray;
    }

    public static JSONObject parseJsonFromArray(JSONArray jsonArray, int index)
    {
        JSONObject json = null;

        try
        {
            json = jsonArray.getJSONObject(index);
        }
        catch (Exception ex)
        {
            ex.printStackTrace();
        }
        return json;
    }

    public static String parseValueByKey(JSONObject json, String key)
    {
        String str = null;

        try
        {
            str = json.getString(key);
        }
        catch (Exception ex)
        {
            ex.printStackTrace();
        }
        return str;
    }
}

File: app/src/main/java/com/ctyeung/popularmoviestage2/data/utilities/JSONhelper.kt
package com.ctyeung.popularmoviestage2.data.utilities

import org.json.JSONArray
import org.json.JSONObject

/**
 * Created by ctyeung on 8/18/17.
 */
object JSONhelper {
    fun parseJson(str: String?): JSONObject? {
        var json: JSONObject? = null
        try {
            json = JSONObject(str)
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
        return json
    }

    fun getJsonArray(json: JSONObject, key: String?): JSONArray? {
        var jsonArray: JSONArray? = null
        try {
            jsonArray = json.getJSONArray(key)
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
        return jsonArray
    }

    fun getJsonArray(jsonString: String?, key: String?): JSONArray? {
        if (null == jsonString) return null
        val json = parseJson(jsonString) ?: return null
        return getJsonArray(json, key)
    }

    fun parseJsonFromArray(jsonArray: JSONArray, index: Int): JSONObject? {
        var json: JSONObject? = null
        try {
            json = jsonArray.getJSONObject(index)
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
        return json
    }

    fun parseValueByKey(json: JSONObject, key: String?): String? {
        var str: String? = null
        try {
            str = json.getString(key)
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
        return str
    }
}
